src/go/k8s/pkg/console/sasl.go

```diff
@@ -184,27 +184,16 @@ func NewKafkaACL(
 
 // Ensure implements Resource interface
 func (k *KafkaACL) Ensure(ctx context.Context) error {
-	// Build ACL for console SASL user to access everything
-	b := kadm.NewACLs().
-		Allow(GenerateSASLUsername(k.consoleobj)).
-		Topics("*").Groups("*").Clusters().Operations(kadm.OpAll).
-		ResourcePatternType(kadm.ACLPatternLiteral)
-	if err := b.ValidateCreate(); err != nil {
-		return fmt.Errorf("validating create ACLs: %w", err)
-	}
-	b.PrefixUserExcept()
-
-	kadmclient, err := k.kafkaAdmin(ctx, k.Client, k.clusterobj, k.store)
+	kadmclient, b, err := k.createAdminClient(ctx)
 	if err != nil {
-		return fmt.Errorf("creating kafka admin client: %w", err)
+		return fmt.Errorf("creating console sasl user: %w", err)
```

Comment:
Nit: this is not creating the user yet

Reply:
the error needs to have context where it is executed. The detail error will be inside `createAdminClient` function.